wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift

```diff
@@ -151,56 +151,70 @@ final class InputBarButtonsView: UIView {
     }
 
     private func layoutAndConstrainButtonRows() {
-        let minButtonWidth: CGFloat = constants.minimumButtonWidth(forWidth: bounds.width)
-
-        guard bounds.size.width >= minButtonWidth * 2 else { return }
+        let minButtonWidth = constants.minimumButtonWidth(forWidth: bounds.width)
 
-        // Drop existing constraints
-        buttons.forEach {
-            $0.roundCorners(edge: .leading)
-            $0.roundCorners(edge: .trailing)
-            $0.removeFromSuperview()
-            buttonInnerContainer.addSubview($0)
+        guard bounds.size.width >= minButtonWidth * 2 else {
+            return
         }
 
-        let ratio = floorf(Float(bounds.width / minButtonWidth))
-        let numberOfButtons: Int = Int(ratio)
-        multilineLayout = numberOfButtons < buttons.count
+        // Reset the container.
+        buttonInnerContainer.removeSubviews()
```

Comment:
What do you think about extracting resetting both container and buttons to a single method?